earth2studio/models/px/microsoft_aurora.py

```diff
@@ -0,0 +1,396 @@
+        self,
+        core_model: torch.nn.Module,
+        z: torch.Tensor,
+        slt: torch.Tensor,
+        lsm: torch.Tensor,
+    ) -> None:
+        super().__init__()
+
+        self.model = core_model
+        self.register_buffer("z", z)
+        self.register_buffer("slt", slt)
+        self.register_buffer("lsm", lsm)
+
+        self._input_coords = OrderedDict(
+            {
+                "batch": np.empty(0),
+                "time": np.empty(0),
+                "lead_time": np.array(
+                    [np.timedelta64(-6, "h"), np.timedelta64(0, "h")]
+                ),
+                "variable": np.array(VARIABLES),
+                "lat": np.linspace(90, -90, 720, endpoint=False),
+                "lon": np.linspace(0, 360, 1440, endpoint=False),
+            }
+        )
+
+        self._output_coords = OrderedDict(
+            {
+                "batch": np.empty(0),
+                "time": np.empty(0),
+                "lead_time": np.array([np.timedelta64(6, "h")]),
+                "variable": np.array(VARIABLES),
+                "lat": np.linspace(90, -90, 720, endpoint=False),
+                "lon": np.linspace(0, 360, 1440, endpoint=False),
+            }
+        )
+        self.device = torch.ones(1).device  # Hack to get default device
+        self.preds_idx = 0
+
+    def input_coords(self) -> CoordSystem:
+        """Input coordinate system of the prognostic model
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        return self._input_coords.copy()
+
+    @batch_coords()
+    def output_coords(
+        self,
+        input_coords: CoordSystem,
+    ) -> CoordSystem:
+        """Output coordinate system of the prognostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system to transform into output_coords
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        output_coords = self._output_coords.copy()
+
+        test_coords = input_coords.copy()
+        test_coords["lead_time"] = (
+            test_coords["lead_time"] - input_coords["lead_time"][-1]
+        )
+        target_input_coords = self.input_coords()
+        for i, key in enumerate(target_input_coords):
+            if key not in ["batch", "time"]:
+                handshake_dim(test_coords, key, i)
+                handshake_coords(test_coords, target_input_coords, key)
+
+        output_coords["batch"] = input_coords["batch"]
+        output_coords["time"] = input_coords["time"]
+
+        output_coords["lead_time"] = (
+            input_coords["lead_time"][-1] + output_coords["lead_time"]
+        )
+
+        return output_coords
+
+    @classmethod
+    def load_default_package(cls) -> Package:
+        """Load prognostic package"""
+        return Package(
+            "/lustre/fsw/coreai_climate_earth2/iauyeung/earth2_aurora",
```

Comment:
I can help you with moving this to a public location

Reply:
@ivanauyeung can you update this to pull from the `microsoft/aurora` huggingface location? For reference you can see how this is done for `Pangu` [here](https://github.com/NVIDIA/earth2studio/blob/e120a36ecf49428c22c1874c7cbd7d60f6e0f7a0/earth2studio/models/px/pangu.py#L205C14-L205C44)